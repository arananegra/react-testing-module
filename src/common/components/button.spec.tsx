import * as React from 'react';
import { fireEvent, render } from '@testing-library/react';
import { Button } from './button';

describe('Button tests', () => {
  it('should render a happy path button with simple props', () => {
    // Arrange
    const props = {
      label: 'Button',
      onClick: jest.fn(),
      className: 'button-style',
    };

    // Act
    const { getByTestId, asFragment } = render(<Button {...props} />);
    const buttonElement = getByTestId('mui-button') as HTMLButtonElement;
    //Assert

    expect(buttonElement).not.toBeUndefined();
    expect(buttonElement.type).toEqual('submit');
    expect(asFragment()).toMatchSnapshot();
  });

  it('should call on click on button when event is fired', () => {
    // Arrange
    const props = {
      label: 'Button',
      onClick: jest.fn(),
      className: 'button-style',
    };

    // Act
    const { getByTestId } = render(<Button {...props} />);
    const buttonElement = getByTestId('mui-button') as HTMLButtonElement;
    fireEvent.click(buttonElement);

    // Assert
    expect(props.onClick).toHaveBeenCalled();
  });

});
