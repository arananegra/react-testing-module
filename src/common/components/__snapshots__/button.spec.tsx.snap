// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button tests should render a happy path button with simple props 1`] = `
<DocumentFragment>
  <button
    class="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedPrimary"
    data-testid="mui-button"
    tabindex="0"
    type="submit"
  >
    <span
      class="MuiButton-label"
    >
      Button
    </span>
    <span
      class="MuiTouchRipple-root"
    />
  </button>
</DocumentFragment>
`;
